### PR TITLE
Some fixes and add User and Group API support

### PR DESCRIPTION
**Pull request by jonnynux**
* Fix Fix exists? check
* Fix detected_by for File
* Fix RateLimitError
* Added User and Group API

### CODE REVIEW

File: test/group_test.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    @api_key  = 'testapikey'
+  end
+
+  def test_class_exists
+    assert VirustotalAPI::Group
+  end
+
+  def test_report_response
+    VCR.use_cassette('group_find') do
+      vturl_report = VirustotalAPI::Group.find(@group_id, @api_key)
```
Comment (crondaemon) [inline test/group_test.rb:17]: vturl_report?

File: test/user_test.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    @api_key = 'testapikey'
+  end
+
+  def test_class_exists
+    assert VirustotalAPI::User
+  end
+
+  def test_report_response
+    VCR.use_cassette('user_find') do
+      vturl_report = VirustotalAPI::User.find(@api_key, @api_key)
```
Comment (crondaemon) [inline test/user_test.rb:16]: Same here.

File: lib/virustotal_api/base.rb [hunk 3]
```diff
@@ -50,7 +50,7 @@ def api_uri
 
     # @return [Boolean] if report for resource exists
     def exists?
-      !report.empty?
+      !!report
```
Comment (crondaemon) [inline lib/virustotal_api/base.rb:57]: What's the deal with the former code? The commit message doesn't really explain why is was buggy and how the new code fixes it.

**Review by crondaemon — CHANGES_REQUESTED** [review]

**Comment by crondaemon** [comment]
LGTM.

**Comment by jonnynux** [comment]
Hi @pwelch, have you had the opportunity to look at this?

Thanks ;)

**Comment by pwelch** [comment]
@jonnynux Looks good to me! 

Should this be cut as a new version or did you have more changes you wanted to roll into the next release?

**Review by pwelch — APPROVED** [review]
👍

**Comment by jonnynux** [comment]
> 
> 
> @jonnynux Looks good to me!
> 
> Should this be cut as a new version or did you have more changes you wanted to roll into the next release?

Yes, new version. At the moment this is enough. I added a commit with the 0.5.2 version, but feel free to edit it as you want.

**Comment by pwelch** [comment]
> > @jonnynux Looks good to me!
> > Should this be cut as a new version or did you have more changes you wanted to roll into the next release?
> 
> Yes, new version. At the moment this is enough. I added a commit with the 0.5.2 version, but feel free to edit it as you want.

Sounds good! I'll Publish it today.

**Comment by pwelch** [comment]
@jonnynux & @crondaemon 

Released v0.5.2: https://rubygems.org/gems/virustotal_api

**Comment by crondaemon** [comment]
:+1: thanks, appreciate.

**Comment by jonnynux** [comment]
Thanks

**Pull request merged**